deploy/operator/before_upgrade.sh

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
```

Comment:
The logic of the upgrade process shouldn't be split between repositories. I can see `before_upgrade.sh` and `after_upgrade.sh` here but not their usage.

Please move https://github.com/openshift/release/pull/21539/files#diff-27e52a9f8e18f6f4a309185c25377a2839112270b55e25429bca81156182bd51R20-R36 logic to be a `./deploy/operator/upgrade.sh` script maintained under the assisted-service repository. This way it would be accessible locally as well.

Reply:
Done. Moved to `./deploy/operator/upgrade.sh`. `before_upgrade.sh` and `after_upgrade.sh` are to be used by CI job. My priority is to get the job ready before feature freeze and then focus on running the upgrade job locally.

Reply:
Great.
Now looking at the CI job, I don't understand why do we need 3 different steps. I'd appreciate if also `after_upgrade` step could be as part of the `upgrade` script.
https://github.com/openshift/release/pull/21539/files#diff-10db92a2ad655eae770c3c0383bf65fa41baf1fe82a8a388e00ea02634e02199R53-R57

The same goes for ZTP. We verify our installation stages as part of the ZTP script.
Then, all the variables will be assigned inside the script and the `after_upgrade.sh` and `before_upgrade.sh` won't be necessairy. In case someone would want to override them - then it should be possible.

Reply:
I suggest making these code style changes later and get the CI job-ready first. I will create a follow-up PR later.